checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,990 @@
+         */
+        @Override
+        public Void visitMethodInvocation(
+                MethodInvocationTree node, AnnotatedTypeMirror annotatedRetType) {
+            AnnotatedTypeMirror receiverType = getReceiverType(node);
+
+            // ReceiverType is null for abstract classes
+            // (Example: Ordering.natural() in tests/all-systems/PolyCollectorTypeVars.java)
+            if (receiverType == null) {
+                return super.visitMethodInvocation(node, annotatedRetType);
+            }
+
+            ExecutableElement m = methodFromUse(node).methodType.getElement();
+
+            // If return type (non-array, non-collection, and non-iterator) resolves to
+            // @OrderNonDet, replaces the annotation on the return type with @NonDet.
+            if (annotatedRetType.getAnnotations().contains(ORDERNONDET)
+                    && !mayBeOrderNonDet(annotatedRetType.getUnderlyingType())) {
+                annotatedRetType.replaceAnnotation(NONDET);
+            }
+
+            // Annotates the return type of "equals()" method called on a Set receiver
+            // as described in the specification of this method.
+
+            // Example1: @OrderNonDet Set<@OrderNonDet List<@Det Integer>> s1;
+            //           @OrderNonDet Set<@OrderNonDet List<@Det Integer>> s2;
+            // s1.equals(s2) is @NonDet
+
+            // Example 2: @OrderNonDet Set<@Det List<@Det Integer>> s1;
+            //            @OrderNonDet Set<@Det List<@Det Integer>> s2;
+            // s1.equals(s2) is @Det
+            // TODO-rashmi: this can be more precise (@Det receiver and @OrderNonDet argument)
+            TypeElement receiverUnderlyingType =
+                    TypesUtils.getTypeElement(receiverType.getUnderlyingType());
+
+            // Without this check, NullPointerException in Collections class with buildJdk.
+            // Likely cause: Collections has a private constructor?
+            // Error at line: public class Collections {
+            // TODO-rashmi: check why?
+            if (receiverUnderlyingType == null) {
+                return super.visitMethodInvocation(node, annotatedRetType);
+            }
+
+            if (isEqualsMethod(m)) {
+                AnnotatedTypeMirror argument = getAnnotatedType(node.getArguments().get(0));
+                if (isSet(receiverUnderlyingType.asType())
+                        && receiverType.hasAnnotation(ORDERNONDET)
+                        && !hasOrderNonDetListAsTypeArgument(receiverType)
+                        && isSet(TypesUtils.getTypeElement(argument.getUnderlyingType()).asType())
+                        && argument.hasAnnotation(ORDERNONDET)
+                        && !hasOrderNonDetListAsTypeArgument(argument)) {
+                    annotatedRetType.replaceAnnotation(DET);
+                }
+            }
+
+            // The following code fixes Issue#14
+            // (https://github.com/t-rasmud/checker-framework/issues/14).
```

Comment:
This issue is still open in the issue tracker, so does this code really fix the issue?  If yes, then close that issue and document why this code is needed.  If not, then this should say it's a "work around" issue 14.